app/src/main/java/org/simple/clinic/appupdate/AppUpdateNotificationWorker.kt

```diff
@@ -93,21 +94,38 @@ class AppUpdateNotificationWorker(
 
     return checkAppUpdateAvailability
         .listen()
-        .filter { it is ShowAppUpdate }
+        .filter { it is ShowAppUpdate || it is DontShowAppUpdate }
         .singleOrError()
         .map { result ->
-          when ((result as ShowAppUpdate).appUpdateNudgePriority) {
-            LIGHT -> showAppUpdateNotification(NOTIFICATION_ID_LIGHT, isLightAppUpdateNotificationShown)
-            MEDIUM -> showAppUpdateNotification(NOTIFICATION_ID_MEDIUM, isMediumAppUpdateNotificationShown)
-            CRITICAL_SECURITY, CRITICAL -> showCriticalUpdateNotification()
-            else -> Result.failure()
+          if (result is ShowAppUpdate) {
```

Comment:
Instead of filtering only `ShowAppUpdate` and `DontShowAppUpdate`, we can instead use exhaustive `when` block to handle the `AppUpdateState`